app/javascript/containers/ReviewModeContainer.jsx

```diff
@@ -50,54 +39,28 @@ function mapCategorytoQuestions(categoryMap, categoryId, building) {
         {...question}
       />
     );
-  })
+  });
 }
 
 class ReviewModeContainer extends React.Component {
-
   constructor(props) {
     super(props);
     this.state = {
-      status_string: "Delegations have not been saved yet.",
-    }
+      status_string: "Delegations have not been saved yet."
+    };
   }
 
   // called when delegation should be submitted
   // should synchronously submit delegations since user expects success
-  async submitDelegation() {
-    var parentQuestionsForDelegations = this.props.questions.filter((question) => {
-      answer = this.props.getAnswer(question.id);
-      return answer && !answer.text && answer.delegation_email;
-    });
-
-    var delegations = [];
-    for (var i = 0; i < parentQuestionsForDelegations.length; i++) {
-      var question = parentQuestionsForDelegations[i];
-      var answer = this.props.getAnswer(question.id);
-      var allDependentQuestions = this.props.getPotentialDependentQuestions(question);
-      allDependentQuestions.push(question);
-
-      allDependentQuestions.map((currentQuestion) => {
-        var currentAnswer = this.props.getAnswer(currentQuestion.id);
-        if (currentAnswer) {
-          var delegation = {
-            email: answer.delegation_email,
-            first_name: answer.delegation_first_name,
-            last_name: answer.delegation_last_name,
-            answer_id: currentAnswer.id,
-          };
-          delegations.push(delegation);
-        }
-      })
-      ;
-    }
-
-    this.setState({status_string: "Saving delegations!"});
-    var success = await postDelegations(delegations);
-    if (success) {
-      this.setState({status_string: "Delegations saved."});
-    } else {
-      this.setState({status_string: "Saving delegations failed. Try again?"});
+  async submitDelegations(delegations) {
+    this.setState({ status_string: "Saving delegations!" });
+    try {
+      let response = await post("/api/delegations", { delegations });
```

Comment:
(See my below comment in `render()` first, but this is the line of code that doesn't need to run if there are no delegations to make)